posts/2021-01-22-ee9-messaging-security-21002-beta.adoc

```diff
@@ -0,0 +1,205 @@
+* Jakarta Expression Language 4.0 (`el-4.0` now `expressionLanguage-4.0`)
+
+[#messaging]
+=== Messaging 3.0
+
+Jakarta Messaging is an API and services that enable applications to create, send, and receive messages via loosely coupled, reliable asynchronous communications. Support for version 3.0 updates the API and services to the jakarta.* namespace and introduces four new features for Jakarta EE 9: `messaging-3.0`, `messagingClient-3.0`, `messagingServer-3.0`, `messagingSecurity-3.0`.
+
+Configuration is the same as the corresponding features for Jakarta EE 8 (`jms-2.0`, `wasJmsClient-2.0`, `wasJmsServer-1.0`, `wasJmsSecurity-1.0`) with one exception: version 3.0 support does not introduce a new feature corresponding to `jmsMdb-1.0`. Instead, use Jakarta EE 9 feature `mdb-4.0` when deploying message-driven beans to connect to MQ providers, JCA resource adapters, or the embedded messaging server.
+
+To enable the new Jakarta EE 9 messaging features, add them to your `server.xml`. Here's a feature configuration that enables applications to access message queues via the Messaging 3.0 API, enables message-driven beans to connect to the embedded messaging server, and enables JNDI lookups for messaging resources.
+
+The following features can be included in your server.xml:
+
+[source, xml]
+----
+ <featureManager>
+       <feature>messagingClient-3.0</feature>
+       <feature>mdb-4.0</feature>
+       <feature>jndi-1.0</feature>
+ </featureManager>
+----
+
+Additional information about the Jakarta Server Faces 3.0 specification can be found link:https://jakarta.ee/specifications/faces/3.0/[here].
+
+[#security]
+=== Security 2.0
+
+The new `appSecurity-4.0` feature has been created to support Jakarta EE 9 and introduces Jakarta Security 2.0 support. The `appSecurityClient-1.0`, `audit-1.0`, `constrainedDelegation-1.0`, `jcaInboundSecurity-1.0`, `jwt-1.0`, `passwordUtilities-1.0` and `spnego-1.0` features have been updated to also support Jakarta EE 9.
+
+The appSecurity-4.0 feature must be added to `server.xml` when using application security with Jakarta EE 9. The other `security-2.0` will automatically adapt to the level of Java EE or Jakarta EE that is already in use, so no change is needed when using them with Jakarta EE 9.
+
+The following features can be included in your server.xml:
+
+[source, xml]
+----
+ <featureManager>
+       <feature>appSecurity-4.0</feature>
+       <feature>appSecurityClient-1.0</feature>
+       <feature>audit-1.0</feature>
+       <feature>constrainedDelegation-1.0</feature>
+       <feature>jcaInboundSecurity-1.0</feature>
+       <feature>jwt-1.0</feature>
+       <feature>passwordUtilities-1.0</feature>
+       <feature>spnego-1.0</feature>
+ </featureManager>
+----
+
+Enable the Jakarta EE 9 beta features in your app's `server.xml`. You can enable the individual features you want or you can just add the Jakarta EE 9 convenience feature to enable all of the Jakarta EE 9 beta features at once:
```

Comment:
So do we need a section about Jakarta Batch and Jakarta XML Web Services and how to enable those features?

Reply:
As far as I know Batch was just a namespace update? No information was provided for either feature so it may need to be added after the blog is published.

Reply:
In the end they all are just namespace updates and package name changes from javax to jakarta.